### ISSUE TITLE
Issue with mutate in 0.7.0

### ISSUE THREAD

**Issue by m0nhawk** [tidyverse/dplyr#2883]
The following code works fine with 0.5.0. But does not in 0.7.0. (found on [StackOverflow](https://stackoverflow.com/q/44597035/1030110))

```
library(dplyr)
tib <- tibble(a = c(1,2,3))

tib %>% mutate(b = a^2, c = sum(b))
```

Output in 0.5.0
```
# A tibble: 3 x 3
      a     b     c
  <dbl> <dbl> <dbl>
1     1     1    14
2     2     4    14
3     3     9    14
```

Output in 0.7.0
```
# A tibble: 3 x 3
      a     b             c
  <dbl> <dbl>         <dbl>
1     1     1 1.482197e-323
2     2     4 1.482197e-323
3     3     9 1.482197e-323
```

**Comment by alistaire47** [comment]
This is a dupe of #2842 fixed by #2869, and currently works as expected on the dev version.

**Comment by lionel-** [comment]
good, thanks!
